spec/factories/course_assessment_submissions.rb

```diff
@@ -22,10 +22,14 @@
     end
 
     trait :graded do
+      transient do
+        with_grade nil
+      end
+
       submitted
       after(:build) do |submission, evaluator|
         submission.answers.each do |answer|
-          answer.grade = Random::DEFAULT.rand(answer.question.maximum_grade)
+          answer.grade = evaluator.with_grade || Random::DEFAULT.rand(answer.question.maximum_grade)
```

Comment:
Maybe if the creation requires custom attributes the answers should be manually created?